cmd/containerd-shim-runhcs-v1/task_hcs.go

```diff
@@ -674,19 +674,20 @@ func hcsPropertiesToWindowsStats(props *hcsschema.Properties) *stats.Statistics_
 
 func (ht *hcsTask) Stats(ctx context.Context) (*stats.Statistics, error) {
 	s := &stats.Statistics{}
-
 	props, err := ht.c.PropertiesV2(ctx, hcsschema.PTStatistics)
-	if err != nil {
+	if err != nil && !isStatsNotFound(err) {
 		return nil, err
 	}
-	if ht.isWCOW {
-		s.Container = hcsPropertiesToWindowsStats(props)
-	} else {
-		s.Container = &stats.Statistics_Linux{Linux: props.Metrics}
+	if props != nil {
```

Comment:
Were we running into an issue where this was nil?

Reply:
Yes, if `PropertiesV2` returns something like a `Not Found` error, since we're ignoring those errors with `isStatsNotFound(err)`, props would be nil

Reply:
No I get that, I guess I'm asking something else so my apologies haha. If we got an `isStatsNotFound` shouldn't we just return `s, nil` and be on our way (empty stats object)? Is there ever a time where props wouldn't be nil if PropertiesV2 returned an error?

Reply:
Well nevermind that wouldn't work in the vm case

Reply:
Had to scroll down a little more 😆